tests/unit/rabbitmq/test_rabbitmq.py

```diff
@@ -109,3 +111,45 @@ async def test_message_generator_10_upstream_error(
             _ = [m async for m in sub.message_generator(propagate_error=False)]
         # would be called by Queue
         self._get_close_mock_fn(mock_con).assert_not_called()
+
+
+class TestUnitRabbitMQURLParsing:
+    """Unit test the URL-parsing by rabbitmq."""
+
+    def test_000(self) -> None:
+        """Sanity check the constants."""
+        assert HUMAN_PATTERN == ("[abc://][USER[:PASS]@]HOST[:PORT][/VIRTUAL_HOST]")
+        assert REGEX_PATTERN == (
+            r"([^:/]+://)?(?P<host>[^:/]*)(:(?P<port>\d+))?(/(?P<virtual_host>.+))?"
+        )
```

Comment:
I don't think you should test for the exact pattern, as this just forces you to update the constant in two places if you ever need to.  You could test for things in the pattern, like `host` or `virtual_host`.  But I'd be wary of putting in too many details, making it difficult to change.  The real test is below anyway.

Reply:
I like these kinds of tests because if we do update the pattern without updating the tests, it'll fail. It's like, "hey! don't forget about the tests!". Now it is just `HUMAN_PATTERN` which isn't terribly useful, but also not terribly unuseful.